plugin/test/node-builder.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { setupServer } from "msw/node";
+import { SourceNodesArgs, NodeInput } from "gatsby";
+
+import { processorMap } from "../src/node-builder"
+
+const server = setupServer();
+
+beforeAll(() => {
+  server.listen();
+});
+
+afterEach(() => {
+  server.resetHandlers();
+});
+
+afterAll(() => {
+  server.close();
+});
+
+describe("When a variant has an image set", () => {
+  const node = {
+    image: {
+      id: "foo1",
+      originalSrc: "https://via.placeholder.com/100x100",
+    },
+    id: "foo2", 
+    internal: {
+      type: "foo3", 
+      contentDigest: "foo4", 
+    }
+  } as NodeInput
+
+  const createNode = jest.fn();
+  const gatsbyApiMock = jest.fn().mockImplementation(() => {
+    return {
+      cache: {
+        set: jest.fn(),
+        get: jest.fn(),
+      },
+      actions: {
+        createNode,
+      },
+      createContentDigest: jest.fn(),
+      createNodeId: jest.fn(),
+      store: jest.fn(),
+      reporter: {
+        info: jest.fn(),
+        error: jest.fn(),
+        panic: jest.fn(),
+        activityTimer: () => ({
+          start: jest.fn(),
+          end: jest.fn(),
+          setStatus: jest.fn(),
+        }),
+      },
+    };
+  });
+
+  const gatsbyApi = gatsbyApiMock as jest.Mock<SourceNodesArgs>;
+
+  describe("and options are set, not to download images", () => {
+    const options = {
+      apiKey: ``,
+      password: ``,
+      storeUrl: "my-shop.shopify.com",
+    };
+
+    it("doesn't create localFile on the node.", async () => {
+      await processorMap.ProductVariant(node, gatsbyApi(), options);
+      expect(node).not.toHaveProperty('localFile');
+    })
+  })
+
+  /*
+  describe("and options are set to download images", () => {
```

Comment:
@sslotsky I didn't get this part to work ...

```
  ● When a variant has an image set › and options are set to download images › creates localFile on the node.

    Failed: "failed to process https://via.placeholder.com/100x100
    TypeError [ERR_INVALID_ARG_TYPE]: The \"path\" argument must be of type string. Received undefined"

      80 |     };
      81 | 
    > 82 |     it("creates localFile on the node.", async () => {
         |     ^
      83 |       await processorMap.ProductVariant(node, gatsbyApi(), options);
      84 |       expect(node).toHaveProperty('localFile');
      85 |     })

      at Env.it (../node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:95:24)
      at Suite.<anonymous> (test/node-builder.spec.ts:82:5)
      at Suite.<anonymous> (test/node-builder.spec.ts:74:3)
```

I was initially thinking to mock more, but have no idea how to mock `processChildImage` and/or `downloadImageAndCreateFileNode` which are directly called from the processor.

Reply:
Got it. Yeah, mocking the exports from `gatsby-source-filesystem` might be the better option here.

Reply:
@kingkero were you going to look into mocking `gatsby-source-filesystem` at all? If you can't get to it, I think it's fine if we jsut remove the failing test here and I can make a ticket to circle back to testing this later. I don't want to block you from being able to use this.

Reply:
@sslotsky Sorry for the late response. We are using this now in production, though currently pointing to my fork where I've already merged both feature requests. It was a bit tough at the end of the week to get everything done, so I didn't have time to look into the open PRs here yet. I did not yet look into mocking `gatsby-source-filesystem`. This week should be a bit more relaxed, so I want to take the time to improve the PRs.

Reply:
No worries. If that's the case, since #91 is in sync with master, I'll invite you to double check it and I'll close this PR.